runhouse/resources/hardware/cluster_factory.py

```diff
@@ -38,8 +38,8 @@ def cluster(
         name (str): Name for the cluster, to re-use later on.
         host (str or List[str], optional): Hostname (e.g. domain or name in .ssh/config), IP address, or list of IP
             addresses for the cluster (the first of which is the head node).
-        ssh_creds (dict, optional): Dictionary mapping SSH credentials.
-            Example: ``ssh_creds={'ssh_user': '...', 'ssh_private_key':'<path_to_key>'}``
+        ssh_creds (dict, optional): SSH credentials, passed as dictionary. Example:
```

Comment:
The original thought behind allowing ssh_creds to receive `Dict` only was to left the API as is. The users are used to pass ssh_creds as dictionaries. Moreover, if the cluster is being created for the first time (and not from_name), we are creating the secret in the cluster factory. 

I think that maybe we should not pass Secrets to the factory method just yet, since we are planning to have CredsSecret class, so probably it will be better to discuss this when we'll develop the new secret. 

@carolineechen what do you think?